scripts/gen-readme.go

```diff
@@ -0,0 +1,252 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"html/template"
+	"os"
+	"os/signal"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+	"syscall"
+)
+
+var (
+	workflowConfigPath string = path.Clean(path.Join("workflow.config.json"))
+	readmeTmplatePath  string = path.Clean(path.Join("templates", "README.tmpl.md"))
+	readmeOutputPath   string = path.Clean(defaultEnv("OUTPUT_PATH", path.Join(".", "README.md")))
+)
+
+// PropertiesConfig are the object proeprties for the *.properties.json files
+type PropertiesConfig struct {
+	Name        string   `json:"name"`
+	Description string   `json:"description"`
+	Creator     string   `json:"creator"`
+	IconName    string   `json:"iconName"`
+	Categories  []string `json:"categories"`
+}
+
+// ReadmeActionConfig is the action template config used for the index README template
+type ReadmeAction struct {
+	Name       string
+	Path       string
+	ReadMePath string
+	Workflows  []ReadmeWorkflow
+}
+
+// ReadmeWorkflowConfig is the workflow config used for the index README template
+type ReadmeWorkflow struct {
+	Name           string
+	RelativeName   string
+	Description    string
+	Starter        bool
+	WorkflowPath   string
+	PropertiesPath string
+}
+
+// ReadmeTemplateConfig is the template config used for the index README template
+type ReadmeTemplateConfig struct {
+	Title   string
+	Actions []ReadmeAction
+}
+
+// Workflow is the object properties for each workflow
+type Workflow struct {
+	Starter        bool   `json:"starter"`
+	Type           string `json:"type"`
+	WorkflowPath   string `json:"workflowPath"`
+	PropertiesPath string `json:"propertiesPath"`
+}
+
+// WorkflowConfig is the object referencing all workflow configs
+type WorkflowConfig map[string]Workflow
+
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+
+	if err := realMain(ctx); err != nil {
+		cancel()
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		os.Exit(1)
+	}
+}
+
+func realMain(ctx context.Context) error {
+	configBytes, err := os.ReadFile(workflowConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to read workflow config file: %w", err)
+	}
+
+	var workflowConfig WorkflowConfig
+	if err := json.Unmarshal(configBytes, &workflowConfig); err != nil {
+		return fmt.Errorf("failed to unmarshal workflow config: %w", err)
+	}
+
+	hasInvalidConfigs := false
+
+	sortedWorkflowsIDs := getSortedWorkflowIDs(workflowConfig)
+
+	readmeActions := map[string]ReadmeAction{}
+
+	for _, workflowID := range sortedWorkflowsIDs {
+		workflow := workflowConfig[workflowID]
+
+		if _, err := os.Stat(workflow.WorkflowPath); os.IsNotExist(err) {
```

Comment:
Splitting these validations into a `validateWorkflow(w Workflow)` would make this more readable